src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexProvider.cs

```diff
@@ -2,27 +2,18 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.WebJobs.Host.Bindings;
 using Microsoft.Azure.WebJobs.Host.Executors;
-using Microsoft.Azure.WebJobs.Host.Storage;
+using Microsoft.Azure.WebJobs.Host.Timers;
 using Microsoft.Azure.WebJobs.Host.Triggers;
 
 namespace Microsoft.Azure.WebJobs.Host.Indexers
 {
     internal class FunctionIndexProvider : IFunctionIndexProvider
     {
-        private readonly ITypeLocator _typeLocator;
-        private readonly ITriggerBindingProvider _triggerBindingProvider;
-        private readonly IBindingProvider _bindingProvider;
-        private readonly IJobActivator _activator;
-        private readonly IFunctionExecutor _executor;
-        private readonly IExtensionRegistry _extensions;
-        private readonly SingletonManager _singletonManager;
-        private readonly TraceWriter _trace;
-
+        private Func<CancellationToken, Task<IFunctionIndex>> _initializeIndex;
```

Comment:
fwiw, I think the generated code here is will be unequivocally worse than before. The compiler will generate an extra class with a closure that does all the same thing that we're removing here.  
I kind of wonder why we're passing all of these types in there in the first place.  


---
In reply to: [109277385](https://github.com/Azure/azure-webjobs-sdk/pull/1100#discussion_r109277385) [](ancestors = 109277385)